### ISSUE TITLE
Help wanted: nixos configure options

### ISSUE THREAD

**Issue by Danack** [Imagick/imagick#399]
@stephank kindly added a docker container for nixos, and the extension builds, but it seems that the array returned by `Imagick::getConfigureOptions();` is quite short.

In particular, it's _really_ suprising to not have the CONFIGURE value returned. The point of the ability to get the configure options is to be able to debug what the configure options were.

I'm not sure if this is actually an issue or not, and will probably close this issue in 3 months if no-one gets around to it but it's slightly disconcerting to have that data not returned. It currently makes test 254_getConfigureOptions.phpt fail, as that test assumes that there will be at least one entry that starts with the letter 'C'.

Examples of output on nixos:

```
array(5) {
  ["DELEGATES"]=>
  string(102) "bzlib cairo djvu fontconfig freetype heic jng jp2 jpeg lcms lzma openexr png rsvg tiff webp x xml zlib"
  ["FEATURES"]=>
  string(28) "Cipher DPC HDRI OpenMP(4.5) "
  ["MAGICK_TEMPORARY_PATH"]=>
  string(4) "/tmp"
  ["NAME"]=>
  string(11) "ImageMagick"
  ["QuantumDepth"]=>
  string(3) "Q16"
}
```

Example output on Debian:

```
array(38) {
  ["CC"]=>
  string(3) "gcc"
  ["CFLAGS"]=>
  string(186) "-I/usr/include/libxml2 -I/usr/include/libpng16 -I/usr/include/freetype2  -fopenmp -Wall -g -O2
  ["CODER_PATH"]=>
  string(56) "/usr/local/lib/ImageMagick-7.0.11/modules-Q16HDRI/coders"
  ["CONFIGURE"]=>
  string(155) "./configure  '--with-quantum-depth=16' '--with-magick-plus-plus=no' '--without-perl' '--disable-static'
  ["CONFIGURE_PATH"]=>
  string(29) "/usr/local/etc/ImageMagick-7/"
  ["COPYRIGHT"]=>
  string(46) "Copyright (C) 1999-2021 ImageMagick Studio LLC"
  ["CPPFLAGS"]=>
  string(34) "-I/usr/local/include/ImageMagick-7"
  ["CXX"]=>
  string(3) "g++"
  ["CXXFLAGS"]=>
  string(9) " -pthread"
  ["DEFS"]=>
  string(15) "-DHAVE_CONFIG_H"
  ["DELEGATES"]=>
  string(33) "freetype jng jpeg png ps xml zlib"
  ["DISTCHECK_CONFIG_FLAGS"]=>
  string(217) " --disable-deprecated  --with-quantum-depth=16  --with-jemalloc=no  --with-umem=no  --with-autotrace=no
  ["DOCUMENTATION_PATH"]=>
  string(34) "/usr/local/share/doc/ImageMagick-7"
  ["EXEC-PREFIX"]=>
  string(10) "/usr/local"
  ["EXECUTABLE_PATH"]=>
  string(14) "/usr/local/bin"
  ["FEATURES"]=>
  string(22) "DPC HDRI Cipher OpenMP"
  ["FILTER_PATH"]=>
  string(57) "/usr/local/lib/ImageMagick-7.0.11/modules-Q16HDRI/filters"
  ["GIT_REVISION"]=>
  string(24) "18571:309fcfa1c:20210328"
  ["HOST"]=>
  string(19) "x86_64-pc-linux-gnu"
  ["INCLUDE_PATH"]=>
  string(32) "/usr/local/include/ImageMagick-7"
  ["LDFLAGS"]=>
  string(17) "-L/usr/local/lib "
  ["LIB_VERSION"]=>
  string(5) "0x70B"
  ["LIB_VERSION_NUMBER"]=>
  string(8) "7,0,11,6"
  ["LIBRARY_PATH"]=>
  string(33) "/usr/local/lib/ImageMagick-7.0.11"
  ["LIBS"]=>
  string(96) "    -lfreetype  -ljpeg    -lpng16                       -lxml2  -lz     -lm    -lpthread  -lgomp"
  ["MAGICK_TEMPORARY_PATH"]=>
  string(4) "/tmp"
  ["NAME"]=>
  string(11) "ImageMagick"
  ["PCFLAGS"]=>
  string(65) "-fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16"
  ["PREFIX"]=>
  string(10) "/usr/local"
  ["QuantumDepth"]=>
  string(2) "16"
  ["RELEASE_DATE"]=>
  string(10) "2021-03-28"
  ["SHARE_PATH"]=>
  string(30) "/usr/local/share/ImageMagick-7"
  ["SHAREARCH_PATH"]=>
  string(48) "/usr/local/lib/ImageMagick-7.0.11/config-Q16HDRI"
  ["TARGET_CPU"]=>
  string(6) "x86_64"
  ["TARGET_OS"]=>
  string(9) "linux-gnu"
  ["TARGET_VENDOR"]=>
  string(2) "pc"
  ["VERSION"]=>
  string(6) "7.0.11"
  ["WEBSITE"]=>
  string(23) "https://imagemagick.org"
}
```

**Comment by stephank** [comment]
This is something to do with ImageMagick trying to read `configure.xml` from FHS paths, but Nix is doing its own thing. We can apparently help it along:

```sh
export MAGICK_CONFIGURE_PATH=/nix/store/dacba3fdfgrmvlxb09s7aakhjdib5z12-imagemagick-7.0.11-8-dev/lib/ImageMagick-7.0.11/config-Q16HDRI
php -d extension=imagick.so -r 'var_dump(Imagick::getConfigureOptions());'
# array(38) { ["CC"]=> string(3) "gcc", ...
```

But that's not really a solution. Regardless, this is an issue in NixOS packaging of ImageMagick, not in Imagick. 🙂

**Comment by Danack** [comment]
Hmm, I guess https://github.com/NixOS/nixpkgs would be the right place to report it?

![image](https://user-images.githubusercontent.com/1505719/118134703-20e4e600-b3fa-11eb-963a-9f8762f21f48.png)

Suspect they might not get round to it for a while...

I think I _won't_ report it. If anyone does care enough to need that info, they can either report it themselves, or do the work required to make it work, but I won't add to their list of stuff.

Thanks for clarifying where the issue is.

**Comment by Danack** [comment]
actuallly, I'll change the test to look for stuff that starts with D as that is present on both debian and nix.

**Comment by stephank** [comment]
Hmm, I did raise the issue in nixpkgs, but it was pointed out that this may not be desirable to fix. Currently, libraries, documentation and headers are split in NixOS, much like other distributions. However, it's a property of how Nix packages things that if the library package were to refer to the other packages, it'd automatically pull them in as dependencies. That defeats the purposes of splitting; not everyone wants those installed all the time.

So instead I added the `MAGICK_CONFIGURE_PATH` variable to our test environment in #400, so it works there at least.
